api/envoy/extensions/filters/http/basic_auth/v3/basic_auth.proto

```diff
@@ -42,3 +42,16 @@ message BasicAuth {
   string forward_username_header = 2
       [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME strict: false}];
 }
+
+message BasicAuthPerRoute {
+  oneof override {
+    option (validate.required) = true;
+
+    // Disable the basic auth filter for this particular vhost or route.
+    // If disabled is specified in multiple per-filter-configs, the most specific one will be used.
+    bool disabled = 1 [(validate.rules).bool.const = true];
+
+    // Username-password pairs for this route.
+    config.core.v3.DataSource users = 2 [(validate.rules).message.required = true, (udpa.annotations.sensitive) = true];
```

Comment:
I think I will have to add a wrapper struct for `users` and `forward_username_header` to make these fields mutually exclusive with `disabled`, right?

Reply:
Do you think that `forward_username_header` should be supported by per-route filter? Or should it be general setting for the filter?